fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt

```diff
@@ -279,22 +284,37 @@ private fun TabItem(
             horizontalArrangement = Arrangement.SpaceBetween,
         ) {
             Row(
-                modifier = Modifier.weight(1f, fill = false),
+                modifier = Modifier.weight(1f),
                 verticalAlignment = Alignment.CenterVertically,
             ) {
+                // This makes sure that isRtl is only calculated when the title changes.
+                val isTitleRtl = remember(state.title) {
+                    BidiFormatter.getInstance().isRtl(state.title)
+                }
+
                 Spacer(modifier = Modifier.size(8.dp))
 
                 TabStripIcon(state.url)
 
                 Spacer(modifier = Modifier.size(8.dp))
 
-                Text(
-                    text = state.title,
-                    color = FirefoxTheme.colors.textPrimary,
-                    maxLines = 1,
-                    overflow = TextOverflow.Ellipsis,
-                    style = FirefoxTheme.typography.subtitle2,
-                )
+                HorizontalFadingEdgeBox(
+                    modifier = Modifier
+                        .weight(1f)
+                        .fillMaxHeight(),
+                    fadeWidth = longerTitleFadeWidth,
+                    backgroundColor = backgroundColor,
+                    isContentRtl = isTitleRtl,
+                ) {
+                    Text(
+                        text = state.title,
+                        modifier = Modifier.align(Alignment.CenterStart),
+                        color = FirefoxTheme.colors.textPrimary,
+                        softWrap = false,
+                        maxLines = 1,
+                        style = FirefoxTheme.typography.subtitle2,
+                    )
```

Comment:
do we need to apply a text direction here as well? I noticed we're doing it in `TabGridItem`, though it's been so long I can't remember why it was done that way.

Reply:
I tried with and without for RTL language website title and it's the same behaviour when the device locale is LTR or RTL.

Although, I think the way we show the fade edge at the end in `HorizontalFadingEdgeBox` could have edge cases where the fade is visible even when the text is actually smaller since fade is on the left side of RTL text. I'll investigate that as a separate issue.